File: Daybreak/Services/Scanner/Models/TitleInfoPayload.cs
namespace Daybreak.Services.Scanner.Models;
public sealed class TitleInfoPayload
{
    public uint CurrentPoints { get; set; }
    public uint PointsNeededNextRank { get; set; }
    public uint TitleId { get; set; }
    public uint TitleTierId { get; set; }
    public uint CurrentTier { get; set; }
    public string? TitleName { get; set; }
    public bool IsPercentageBased { get; set; }
}

File: Daybreak.GWCA/source/payloads/TitleInfoPayload.cpp
#include <cstdint>
#include <json.hpp>
#include <payloads/TitleInfoPayload.h>

using json = nlohmann::json;

namespace Daybreak {
    void to_json(json& j, const TitleInfoPayload& p) {
        j = json
        {
            {"TitleId", p.TitleId},
            {"TitleTierId", p.TitleTierId},
            {"TitleName", p.TitleName},
            {"CurrentPoints", p.CurrentPoints},
            {"PointsNeededNextRank", p.PointsNeededNextRank},
            {"IsPercentageBased", p.IsPercentageBased},
            {"CurrentTier", p.CurrentTier}
        };
    }
}